cmd/api/server.go

```diff
@@ -3,17 +3,35 @@ package main
 import (
 	"net/http"
 	"time"
+
+	"github.com/rs/cors"
 )
 
+func (app *application) configure() http.Handler {
```

Comment:
In Go, we create middlewares with http.HandlerFunc adapter. You can create a method that takes http.Handler and returns http.Handler covered with http.HandlerFunc. 

So in app.routes we can return app.enableCors(mux). And we wouldn't need to change other code (In app.routes, calling app.enableCors(mux) applies CORS across all routes served by mux, keeping the setup clean and centralized. 

In app.routes, calling app.enableCors(mux) applies CORS across all routes served by mux, keeping the setup clean and centralized.

Reply:
We can leave it for now, but we will have a lot of others middlewares and shoukd switch to http.HandlerFunc

Reply:
And I think it would be good to keep our middlewares in separate file.